Gems/Atom/RPI/Code/Source/RPI.Edit/Material/MaterialTypeSourceData.cpp

```diff
@@ -112,89 +127,351 @@ namespace AZ
+                {
+                    return existingProperty->GetName() == name;
+                });
+
+            if (iter != m_properties.end())
+            {
+                AZ_Error("Material source data", false, "PropertyGroup name '%.*s' collides with a Property of the same name", AZ_STRING_ARG(name));
+                return nullptr;
+            }
+
+            return AddPropertyGroup(name, m_propertyGroups);
+        }
+        
+        MaterialTypeSourceData::PropertyGroup* MaterialTypeSourceData::AddPropertyGroup(AZStd::string_view propertyGroupId)
+        {
+            AZStd::vector<AZStd::string_view> splitPropertyGroupId = SplitId(propertyGroupId);
+
+            if (splitPropertyGroupId.size() == 1)
+            {
+                return PropertyGroup::AddPropertyGroup(propertyGroupId, m_propertyLayout.m_propertyGroups);
+            }
+
+            PropertyGroup* parentPropertyGroup = FindPropertyGroup(splitPropertyGroupId[0]);
+            
+            if (!parentPropertyGroup)
+            {
+                AZ_Error("Material source data", false, "PropertyGroup '%.*s' does not exists", AZ_STRING_ARG(splitPropertyGroupId[0]));
+                return nullptr;
+            }
+
+            return parentPropertyGroup->AddPropertyGroup(splitPropertyGroupId[1]);
+        }
+        
+        MaterialTypeSourceData::PropertyDefinition* MaterialTypeSourceData::AddProperty(AZStd::string_view propertyId)
+        {
+            AZStd::vector<AZStd::string_view> splitPropertyId = SplitId(propertyId);
+
+            if (splitPropertyId.size() == 1)
+            {
+                AZ_Error("Material source data", false, "Property id '%.*s' is invalid. Properties must be added to a PropertyGroup (i.e. \"general.%.*s\").", AZ_STRING_ARG(propertyId), AZ_STRING_ARG(propertyId));
+                return nullptr;
+            }
+
+            PropertyGroup* parentPropertyGroup = FindPropertyGroup(splitPropertyId[0]);
+            
+            if (!parentPropertyGroup)
+            {
+                AZ_Error("Material source data", false, "PropertyGroup '%.*s' does not exists", AZ_STRING_ARG(splitPropertyId[0]));
+                return nullptr;
+            }
 
-        const MaterialTypeSourceData::GroupDefinition* MaterialTypeSourceData::FindGroup(AZStd::string_view groupName) const
+            return parentPropertyGroup->AddProperty(splitPropertyId[1]);
+        }
+        
+        const MaterialTypeSourceData::PropertyGroup* MaterialTypeSourceData::FindPropertyGroup(AZStd::span<const AZStd::string_view> parsedPropertyGroupId, AZStd::span<const AZStd::unique_ptr<PropertyGroup>> inPropertyGroupList) const
         {
-            for (const GroupDefinition& group : m_propertyLayout.m_groups)
+            for (const auto& propertyGroup : inPropertyGroupList)
             {
-                if (group.m_name == groupName)
+                if (propertyGroup->m_name != parsedPropertyGroupId[0])
```

Comment:
maybe this function should check `parsedPropertyGroupId` (input parameter) is no empty?